### ISSUE TITLE
Workspaces

### ISSUE THREAD

**Issue by ackzell** [jsreport/playground#2]
The playground has a feature for "grouping" the different parts of a report. Is this called a workspace?

When on the playground you click on "new template" the GUI presents you with a sort of a clean state so that you can start creating assets, adding templates, images, etc.. 

Also this menu:
![image](https://cloud.githubusercontent.com/assets/601266/26705948/3171d74c-46ff-11e7-88c7-eae88dfe3ef9.png)

Is it possible to have that in a local jsreport studio?

**Comment by pofider** [comment]
The "workspaces" is jsreport extension implemented only for the playground so far. It is implemented as extension because there is potential that it could be useful in other scenarios and we could release it with guidelines as official one. 

Can you describe a bit your scenario so we can learn how it could be used?

In every case feel free to take the code from this repository and use it as you need.

**Comment by ackzell** [comment]
Yes, I can think about having more than 1 template for a report, and having all the assets, scripts, images would be too distracting and maybe somewhat hard to find which part belongs where.

I just recently learned about _tags_ and being able to filter by a particular one. Which does kind of solve the problem.

Although I like the approach of having the workspaces: I can imagine if a jsreport Studio instance were shared (say between teams in a company), they could "activate" their own workspace and not care about other people's reports.

I hope that makes sense.

**Comment by bjrmatos** [comment]
> Yes, I can think about having more than 1 template for a report, and having all the assets, scripts, images would be too distracting and maybe somewhat hard to find which part belongs where.



> Although I like the approach of having the workspaces: I can imagine if a jsreport Studio instance were shared (say between teams in a company), they could "activate" their own workspace and not care about other people's reports.

yes!! when i was working with other teams, we really missed this feature, different teams/people working in different reports for different clients.. now that i know that it exists i also think that an workspace extension makes a lot of sense.

**Comment by bjrmatos** [comment]
a big 👍 for publishing the workspace code as an extension
